### PR TITLE
Update CI rubies and update gems in `Gemfile.lock`

### PR DESCRIPTION
**Pull request by shioyama**
I was getting some failures locally because of missing `attributes_in_database` which I think was added in AR 7, so I bumped all the gems in `Gemfile.lock` and also bumped rubies in `ci.yml` to add 3.1.

### CODE REVIEW

**Comment by shioyama** [comment]
I'm actually wondering if this repo should even have a `Gemfile.lock`. Usually for gems it's more of a nuisance than anything.

**Comment by casperisfine** [comment]
> I'm actually wondering if this repo should even have a `Gemfile.lock`. Usually for gems it's more of a nuisance than anything.

That's the recommended thing this days. Otherwise you can't bisect and different contributors have random versions of gems.

What is recommended too however is to have a CI job that does delete the lock and bundle again to test with the newest available dependencies. (or to setup dependabot).

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ jobs:
       - name: Set up Ruby
         uses: ruby/setup-ruby@v1
         with:
-          ruby-version: '2.6'
+          ruby-version: '3.1'
```
Comment (casperisfine) [inline .github/workflows/ci.yml:16]: `rubocop` should be tested with the oldest supported Ruby.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -23,7 +23,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        ruby: [ ruby-head, '3.0', '2.7', '2.6' ]
+        ruby: [ ruby-head, '3.1', '3.0', '2.7' ]
```
Comment (casperisfine) [inline .github/workflows/ci.yml:26]: Why the `2.6` drop? I added it to support latest jruby.

**Comment by casperisfine** [comment]
> to drop 2.6

We can't do that, we have one jruby app, and latest jruby is syntax compatible with ruby 2.6.

We could drop 2.6, but then we should add jruby instead.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (shioyama) [reply .github/workflows/ci.yml:26]: Oh I can add it back, I thought it was unsupported.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (shioyama) [reply .github/workflows/ci.yml]: Ok reverted this 👍

**Pull request merged**